Create Azure Purview workbook

libarson opened the pull request:
Fixes #

## Proposed Changes

  -
  -
  -

saasri commented on Workbooks/Azure Purview:
```diff
@@ -0,0 +1,1497 @@
+    {
+      "type": 1,
+      "content": {
+        "json": "__Azure Purview__\r\n\r\nAzure Purview is a unified data governance service that helps you manage and govern your on-premises, multicloud, and software-as-a-service (SaaS) data. It creates a holistic, up-to-date map of your data landscape with automated data discovery, sensitive data classification, and end-to-end data lineage that empowers you to find valuable and trustworthy da ...
```
'on-prem'

saasri commented on Workbooks/Azure Purview:
```diff
@@ -0,0 +1,1497 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "\r\nlet classifiedAssets = PurviewDataSensitivityLogs\r\n| where ActivityType == \"Classification\" \r\n| where \"{PurviewAccount:label}\" == \"All\" or PurviewAccountName in~ (split(\"{PurviewAccount:label}\", \", \"))\r\n| where \"{DataSource:label}\" == \"All\" or SourceType in~ (split(\"{DataSource:label}\", \", \"))\r\n| extend CollectionName = iff(SourceCollectionName == \ ...
```
'newSensitivityLabelName' looks off, just 'LabelName' works, you should try to keep this consistent across queries. 'AssetCreationTime' is missing

v-rucdu commented:
@libarson : For deploying any workbook its entry needs to be added in [WorkbooksMetadata.json](https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json).

Please refer [instructions](https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/README.md) to understand more on how to add a new workbook.

v-rucdu commented on Workbooks/AzurePurview.json:
```diff
@@ -0,0 +1,1452 @@
+      }
+    }
+  ],
+  "fallbackResourceIds": [
```
Please remove the reference to Azure resource by making the fallbackResourceIds attribute empty as below.
"fallbackResourceIds": [],

Thanks,
Ruchita

v-rucdu commented:
Please provide atleast 2 images one for white theme and other for black theme.

libarson commented:
@v-rucdu I added in the black/white images along with entry to the WorkbooksMetadata.json file. I also removed the reference in fallbackResourceIds. 

I see that some of the checks are failing, can you let me know what I need to update to allow them to pass?

v-rucdu commented on Workbooks/WorkbooksMetadata.json:
```diff
@@ -1415,5 +1415,18 @@
+      "description": "This workbook provides insights into the sensitivity of data that has been scanned by Azure Purview, including classifications and labels found across the data estate.",
+      "dataTypesDependencies": ["PurviewDataSensitivityLogs"],
+      "dataConnectorsDependencies": ["AzurePurview"],
+      "previewImagesFileNames": ["AzurePurviewWhite1.png", "AzurePurviewBlack1.png", "AzurePurviewWhite2.png", "AzurePurviewBlack2.png","AzurePurviewWhite3.png", "AzurePurviewBlack3.png"],
```
Please maintain consistency in the file extensions, while creating and mentioning it in Metadata file. This is one of the reason for validation failure.

e.g. File is added with name as  **AzurePurviewWhite1.PNG** under preview folder and is being referenced as **AzurePurviewWhite1.png** in metadata file. 

Please fix it for all the images.

Thanks!!!

libarson replied:
Fixed the file extensions for the images in the preview folder.

libarson commented on Workbooks/Images/Logos/AzurePurview.svg:
```diff
@@ -0,0 +1,33 @@
+<stop stop-color="#29BADE"/>
+<stop offset="1" stop-color="#50E6FF"/>
+</linearGradient>
+<linearGradient id="paint1_linear" x1="6.446" y1="14.626" x2="6.446" y2="3.374" gradientUnits="userSpaceOnUse">
```
Updated the svg IDs to unique GUIDs.

v-rucdu commented on Workbooks/WorkbooksMetadata.json:
```diff
@@ -1415,5 +1415,18 @@
+      "workbookKey": "AzurePurview",
+      "logoFileName": "AzurePurview.svg",
+      "description": "This workbook provides insights into the sensitivity of data that has been scanned by Azure Purview, including classifications and labels found across the data estate.",
+      "dataTypesDependencies": ["PurviewDataSensitivityLogs"],
```
DataType should be the table name, so please change it to 
**AzureDiagnostics**

Thanks!!!

libarson replied:
DataTypesDepdendencies changed to AzureDiagnostics.

v-rucdu commented on Workbooks/WorkbooksMetadata.json:
```diff
@@ -1415,5 +1415,18 @@
+      "logoFileName": "AzurePurview.svg",
+      "description": "This workbook provides insights into the sensitivity of data that has been scanned by Azure Purview, including classifications and labels found across the data estate.",
+      "dataTypesDependencies": ["PurviewDataSensitivityLogs"],
+      "dataConnectorsDependencies": ["AzurePurview"],
```
DataConnectorDependencies should have the value of  **id** attribute from DataConnector json file.

libarson replied:
DataConnectorDependencies has been updated to have the id attribute for the data connector.
